### PR TITLE
Cr 11553

### PR DESCRIPTION
**Pull request by kim-codefresh**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -254,8 +254,6 @@ github.com/cockroachdb/datadriven v0.0.0-20200714090401-bf6692d28da5/go.mod h1:h
 github.com/cockroachdb/errors v1.2.4/go.mod h1:rQD95gz6FARkaKkQXUksEje/d9a6wBJoCr5oaCLELYA=
 github.com/cockroachdb/logtags v0.0.0-20190617123548-eb05cc24525f/go.mod h1:i/u985jwjWRlyHXQbwatDASoW0RMlZ/3i9yJHE2xLkI=
 github.com/codahale/hdrhistogram v0.0.0-20161010025455-3a0bb77429bd/go.mod h1:sE/e/2PUdi/liOCUjSTXgM1o87ZssimdTWN964YiIeI=
```
Comment (kim-codefresh) [inline go.sum:256]: update go sdk version once it released

File: cmd/commands/runtime.go [hunk 2]
```diff
@@ -226,6 +227,7 @@ func NewRuntimeInstallCommand() *cobra.Command {
 				"Runtime name":              installationOpts.RuntimeName,
 				"Repository URL":            installationOpts.InsCloneOpts.Repo,
 				"Ingress host":              installationOpts.IngressHost,
+				"Internal ingress host":     installationOpts.InternalIngressHost,
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: Please add this only if it's not empty string.

File: cmd/commands/runtime.go [hunk 3]
```diff
@@ -434,7 +437,11 @@ func ensureIngressHost(cmd *cobra.Command, opts *RuntimeInstallOptions) error {
 		}
 	}
 
-	parsed, err := url.Parse(opts.IngressHost)
+	host := opts.IngressHost
+	if opts.InternalIngressHost != "" {
+		host = opts.InternalIngressHost
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: I don't think it's a good idea to use internal ingress as the external ingress if external is not provided.

File: pkg/runtime/runtime.go [hunk 4]
```diff
@@ -51,15 +51,16 @@ type (
...
-		IngressController  string          `json:"ingressController"`
-		Repo               string          `json:"repo"`
+		DefVersion          *semver.Version `json:"defVersion"`
+		Version             *semver.Version `json:"version"`
+		BootstrapSpecifier  string          `json:"bootstrapSpecifier"`
+		Components          []AppDef        `json:"components"`
+		Cluster             string          `json:"cluster"`
+		IngressHost         string          `json:"ingressHost"`
+		IngressClass        string          `json:"ingressClassName"`
+		InternalIngressHost string          `json:"InternalIngressHost"`
```
Comment (roi-codefresh) [inline pkg/runtime/runtime.go]: bug with capital `I` in the beginning

Comment (roi-codefresh) [reply pkg/runtime/runtime.go]:
```suggestion
		InternalIngressHost string          `json:"internalIngressHost"`
```

**Review by roi-codefresh — APPROVED** [review]

File: cmd/commands/runtime.go [hunk 5]
```diff
@@ -245,6 +253,7 @@ func NewRuntimeInstallCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&installationOpts.IngressHost, "ingress-host", "", "The ingress host")
 	cmd.Flags().StringVar(&installationOpts.IngressClass, "ingress-class", "", "The ingress class name")
+	cmd.Flags().StringVar(&installationOpts.InternalIngressHost, "internal-ingress-host", "", "The internal ingress host (Optional)")
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]:
```suggestion
	cmd.Flags().StringVar(&installationOpts.InternalIngressHost, "internal-ingress-host", "", "The internal ingress host (Optional, by default the external ingress will be used for both internal and external traffic)")
```

File: cmd/commands/runtime.go [hunk 6]
```diff
@@ -258,6 +267,8 @@ func NewRuntimeInstallCommand() *cobra.Command {
 	cmd.Flags().BoolVar(&store.Get().SetDefaultResources, "set-default-resources", false, "If true, will set default requests and limits on all of the runtime components")
 	cmd.Flags().BoolVar(&installationOpts.FromRepo, "from-repo", false, "Installs a runtime from an existing repo. Used for recovery after cluster failure")
 	cmd.Flags().StringToStringVar(&installationOpts.NamespaceLabels, "namespace-labels", nil, "Optional labels that will be set on the namespace resource. (e.g. \"key1=value1,key2=value2\"")
+	cmd.Flags().StringToStringVar(&installationOpts.InternalIngressAnnotation, "internal-ingress-annotation", nil, "Optional internall ingress host annotation")
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go:270]:
```suggestion
	cmd.Flags().StringToStringVar(&installationOpts.InternalIngressAnnotation, "internal-ingress-annotation", nil, "Optional internal ingress host annotation")
```

Comment (roi-codefresh) [inline cmd/commands/runtime.go]:
```suggestion
	cmd.Flags().StringToStringVar(&installationOpts.InternalIngressAnnotation, "internal-ingress-annotation", nil, "Add annotations to the internal ingress")
```

File: cmd/commands/runtime.go [hunk 7]
```diff
@@ -258,6 +267,8 @@ func NewRuntimeInstallCommand() *cobra.Command {
...
+	cmd.Flags().StringToStringVar(&installationOpts.ExternalIngressAnnotation, "external-ingress-annotation", nil, "Optional external ingress host annotation")
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]:
```suggestion
	cmd.Flags().StringToStringVar(&installationOpts.ExternalIngressAnnotation, "external-ingress-annotation", nil, "Add annotations to the external ingress")
```

File: cmd/commands/runtime.go [hunk 8]
```diff
@@ -245,6 +253,7 @@ func NewRuntimeInstallCommand() *cobra.Command {
 
 	cmd.Flags().StringVar(&installationOpts.IngressHost, "ingress-host", "", "The ingress host")
 	cmd.Flags().StringVar(&installationOpts.IngressClass, "ingress-class", "", "The ingress class name")
+	cmd.Flags().StringVar(&installationOpts.InternalIngressHost, "internal-ingress-host", "", "The internal ingress host (Optional, by default the external ingress will be used for both internal and external traffic)")
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]:
```suggestion
	cmd.Flags().StringVar(&installationOpts.InternalIngressHost, "internal-ingress-host", "", "The internal ingress host (by default the external ingress will be used for both internal and external traffic)")
```

File: cmd/commands/runtime.go [hunk 9]
```diff
@@ -1881,6 +1935,12 @@ func createWorkflowsIngress(ctx context.Context, opts *RuntimeInstallOptions, rt
 	return apu.PushWithMessage(ctx, r, "Created Workflows Ingress")
 }
 
+func mergeAnnotation(annotation map[string]string, newAnnotation map[string]string) {
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]:
```suggestion
func mergeAnnotations(annotation map[string]string, newAnnotation map[string]string) {
```

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**